event-driven-generated/src/event.rs

```diff
@@ -0,0 +1,90 @@
+use ratatui::crossterm::event::{self, Event as CrosstermEvent};
+use std::sync::mpsc;
+use std::thread;
+use std::time::{Duration, Instant};
+
+/// Application events.
+///
+/// You can extend this enum with your own custom events.
+#[derive(Clone, Debug)]
+pub enum AppEvent {
+    /// Custom increment event.
+    Increment,
+    /// Custom decrement event.
+    Decrement,
+    /// Quit the application.
+    Quit,
+}
+
+/// Representation of all possible events.
+#[derive(Clone, Debug)]
+pub enum Event {
+    /// An event that is emitted on a regular schedule.
+    ///
+    /// Use this event to run any code which has to run outside of being a direct response to
+    /// a user event. e.g. polling exernal systems, updating animations, or rendering the UI
+    /// based on a fixed frame rate.
+    Tick,
+    /// Crossterm events.
+    ///
+    /// These events are emitted by the terminal.
+    Crossterm(CrosstermEvent),
+    /// Application events.
+    ///
+    /// Use this event to emit custom events that are specific to your application.
+    App(AppEvent),
+}
+
+/// Terminal event handler.
+#[derive(Debug)]
+pub struct EventHandler {
+    /// Event sender channel.
+    sender: mpsc::Sender<Event>,
+    /// Event receiver channel.
+    receiver: mpsc::Receiver<Event>,
+}
+
+impl EventHandler {
+    /// Constructs a new instance of [`EventHandler`].
+    pub fn new() -> Self {
+        let tick_rate = Duration::from_secs_f32(1.0 / 30.0);
```

Comment:
I don't mind a small comment on this one, though a constant in this module might be best (e.g. `const TICK_FPS: f64 = 30.0`)

```suggestion
        // send a tick event at 30fps
        let tick_rate = Duration::from_secs_f32(1.0 / 30.0);
```